src/storage/src/liquid_array/primitive_array.rs

```diff
@@ -142,53 +142,124 @@ where
     }
 
     /// Create a Liquid primitive array from an Arrow primitive array.
-    pub fn from_arrow_array(arrow_array: PrimitiveArray<T>) -> LiquidPrimitiveArray<T> {
-        let min = match arrow::compute::kernels::aggregate::min(&arrow_array) {
-            Some(v) => v,
-            None => {
-                // entire array is null
+    pub fn from_arrow_array(
+        arrow_array: PrimitiveArray<T>,
+        encoding: u8,
+    ) -> LiquidPrimitiveArray<T> {
+        if encoding == 0 {
```

Comment:
It's probably fine for a quick prototype, but usually we'd like to avoid using magic numbers.

I suggest we create a `LiquidPrimiveDeltaArray<T>` here and implement the `from_arrow_array` without using a extra encoding parameter.

This will also make your development easier because it won't interfere with existing code.